client/src/util/exponentialBackoffFetch.js

```diff
@@ -1,68 +1,56 @@
 export default function createBackoffFetch(retryTimes = 3) {
   let prevHash = null;
   let retryCount = 0;
-  let response = null;
```

Comment:
since we already capture `promise` and `resolve/reject` it accordingly, the `promise` will store its `success/error` state for us!